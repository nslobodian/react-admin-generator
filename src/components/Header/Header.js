import React from 'react'
import Link from 'react-router/lib/Link'
import cn from 'classnames'
import urls from 'utils/urls'
import NavDropdown from 'react-bootstrap/lib/NavDropdown'
import { compose, withHandlers, withState } from 'recompose'
import BlueMen from 'static/images/Aiesec-white-blue.png'
const {
  members, home, projects,
} = urls

export const CustomLi = ({ active, onSelect, ...props }) => <li onClick={onSelect} {...props} />

const enhance = compose(
  withState('collapse', 'changeCollapsedStatus', false),
  withHandlers({
    toggleStatus: props => () => {
      props.changeCollapsedStatus(collapse => !collapse)
    },
  })
)
export const Header = enhance(({ loggedIn, logout, username, roles, toggleStatus, collapse, ordersVisible,
  usersVisible, inspirationsVisible, inspirationsEditable, productsVisible, erpVisible, erpEditable }) =>
  loggedIn ? <nav className='navbar navbar-inverse'>
    <div className='container-fluid'>
      <div className='navbar-header'>
        <button type='button' className='navbar-toggle collapsed' data-toggle='collapse'
          data-target='#bs-example-navbar-collapse-1' aria-expanded='false' onClick={toggleStatus}>
          <span className='sr-only'>Toggle navigation</span>
          <span className='icon-bar' />
          <span className='icon-bar' />
          <span className='icon-bar' />
        </button>
        <Link to={home} className='navbar-brand'><img className='img-responsive' style={{ width: 150 }} src={BlueMen} alt='blueman' /></Link>
      </div>

      <div className={cn('collapse navbar-collapse', collapse && 'in')}>
        <ul className='nav navbar-nav'>
          <li>
            <Link to={members}>Members</Link>
          </li>
          <li>
            <Link to={projects}>Projects</Link>
          </li>
          {/* <NavDropdown title='Notifications' id='nav-dropdown-notif'>
            <CustomLi><Link to={notifications}>Published Notifications</Link></CustomLi>
            <CustomLi><Link to={createNotifications}>Create Notification</Link></CustomLi>
          </NavDropdown> */}
        </ul>
        <ul className='nav navbar-nav navbar-right'>
          <li>
            <a href='#'>{username}</a>
          </li>
          <li>
            <button onClick={logout} className='btn btn-default navbar-btn'>Logout</button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
    : null
)

export default Header
