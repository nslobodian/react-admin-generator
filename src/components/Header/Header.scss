@import '~styles/shared';

.Header {
  height: 70px;
  border: none;
  border-radius: 0;

  background-color: $dark-blue;

  .navbarBrand {
    float: left;
    height: 50px;
    padding-left: 10px;

    .Logo {
      display: block;

      svg {
        width: 100%;
        height: 70px;
        fill: #fff;
      }
    }
  }

  .logoutButton {
    @include button-variant(#fff, $pink, $pink);
    @include button-size(8px, 12px, 14px, 24px, 5px);
  }

  .navbarRight {
    margin-right: 0;
  }
}
