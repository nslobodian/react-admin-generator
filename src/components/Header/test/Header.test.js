import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Header, CustomLi }from '../Header'

const props = {
  loggedIn: true,
  logout: jest.fn(),
  username: 'Username',
  toggleStatus: jest.fn(),
  collapse: true,
  resetCreatePurchaseOrder: jest.fn(),
}

describe('(Components) Header', () => {
  it('Should render component properly - default', () => {
    const wrapper = shallow(
      <Header
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  }),
  it('Should render component properly - default', () => {
    const active = true;
    const onSelect = jest.fn();
    const wrapper = shallow(
      <CustomLi
        {...props}
        active={active}
        onSelect={onSelect}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

