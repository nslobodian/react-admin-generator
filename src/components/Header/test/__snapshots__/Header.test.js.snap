// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) Header Should render component properly - default 1`] = `
<_class
  changeCollapsedStatus={[Function]}
  collapse={false}
  loggedIn={true}
  logout={[Function]}
  resetCreatePurchaseOrder={[Function]}
  toggleStatus={[Function]}
  username="Username"
/>
`;

exports[`(Components) Header Should render component properly - default 2`] = `
<li
  collapse={true}
  loggedIn={true}
  logout={[Function]}
  onClick={[Function]}
  resetCreatePurchaseOrder={[Function]}
  toggleStatus={[Function]}
  username="Username"
/>
`;
